### PR TITLE
feat: payments history for current user

### PR DESCRIPTION
**Pull request by traderstechie**
## Description:
A feature that fetches and returns all payments for a currently logged-in user, in a paginated manner. It's a protected endpoint at `/api/v1/payments/current-user`.

## Related Issue:

[[FEAT]: Payments History For Current User #500](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/500)

## Motivation and Context
To provide users with a means to visualise their payments history once they are properly authenticated.

## How Has This Been Tested?

- [x] Unittest with pytest.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:
 
- [x]  My code follows the code style of this project.
- [x]  I have updated the documentation accordingly.
- [x]  I have read the CONTRIBUTING document.
- [x]  I have added tests to cover my changes.
- [x]  All new and existing tests passed.

### CODE REVIEW

File: api/v1/services/payment.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        new_payment = Payment(**schema.model_dump())
+        db.add(new_payment)
+        db.commit()
+        db.refresh(new_payment)
+
+        return new_payment
+    
+
+    def fetch_all(self, db: Session, **query_params: Optional[Any]):
+        '''Fetch all payments with option tto search using query parameters'''
```
Comment (Goketech) [inline api/v1/services/payment.py:25]: fix the typo in the comment

**Review by Goketech — CHANGES_REQUESTED** [review]
Nice work! Fix the little typo issue

File: api/v1/routes/payments.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    db: Session = Depends(get_db), 
+):
+    '''
+    Endpoint to retrieve a paginated list of payments of ``current_user``.
+
+    Query parameter: 
+        - limit: Number of payment per page (default: 10, minimum: 1)
+        - page: Page number (starts from 1)
+    '''
+    if not current_user:
```
Comment (joboy-dev) [inline api/v1/routes/payments.py:28]: There is no need for this check. Remove it

File: api/v1/services/payment.py
Comment (joboy-dev) [inline api/v1/services/payment.py:1]: Great work on using services

**Review by joboy-dev — COMMENTED** [review]
Make required changes and I will merge

File: api/v1/services/payment.py (hunk 1 above)
Comment (traderstechie) [reply api/v1/services/payment.py]: Fixed @Goketech

File: api/v1/routes/payments.py (hunk 2 above)
Comment (traderstechie) [reply api/v1/routes/payments.py]: Removed @joboy-dev

File: api/v1/services/payment.py
Comment (traderstechie) [reply api/v1/services/payment.py:1]: Thank you, chief! 🙏

**Comment by Goketech** [comment]
fix your conflicts

**Comment by traderstechie** [comment]
Conflicts fixed @Goketech

**Comment by traderstechie** [comment]
And, thanks for the nice comment earlier @Goketech

**Review by SundayMba — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Pull request merged**